test/cdd/CDDTest.java

```diff
@@ -54,7 +55,7 @@ public void testConjunctionSameTypeWithOverlap() throws CddNotRunningException,
         guardList.add(g3);
         guardList.add(g4);
 
-        System.out.println(new CDD(new AndGuard(guardList)).getGuard(clocks));
+        Log.debug(new CDD(new AndGuard(guardList)).getGuard(clocks));

```

Comment:
This is a very good point. There is 1-2 function calls from a ``Log.debug`` before we check whether it should be logged. I think the solution to this is *lazy evaluation* where we use ``Supplier``. A rough draft could be something along the lines of:
```
    public static void debug(Supplier<String> supplier) {
        if (urgency.level >= Urgency.Debug.level) {
            out(format(supplier.get(), Urgency.Debug));
        }
    }

    // Calling it would look like the following:
   Log.debug(() -> "Hello, World!");
```
Here the ``Supplier```is only evaluated if we need its value. I think a new issue on this topic should be opened.

Reply:
> I think a new issue on this topic should be opened.

Could you open a new issue on this? I'm missing the GitLab function to resolve a conversation (thread) by creating an issue.

Reply:
The [issue](https://github.com/Ecdar/j-Ecdar/issues/77#issue-1408270656) has now been created. Would you please verify that it is correct, before i resolve this comment?